### ISSUE TITLE
How do i catch socket exceptions

### ISSUE THREAD

**Issue by patrickoramah** [CodingAleCR/http_interceptor#34]
I am trying to catch Socket exceptions in my service but it keeps slipping through and throwing the Exception block instead.

This is my implementation. Kindly advise
```
    try {
      http.Response response =
          await client.post('$baseUrl/auth/', body: login.toJson());
      Map<String, dynamic> data = jsonDecode(response.body);
      if (response.statusCode == 401) {
        throw Failure('Wrong username and password combination');
      }
      return data;
    } on HttpException catch (e) {
      throw Failure(e.toString());
    } on SocketException {
      throw Failure('No Internet connection 😑');
    } on FormatException {
      throw Failure('Bad response format 👎');
    } on Exception {
      throw Failure('Unexpected error 😢');
    }
```

```
class HttpInterceptor implements InterceptorContract {
  @override
  Future<RequestData> interceptRequest({RequestData data}) async {
    if (data.headers.containsKey('requireAuth')) {
      String accessToken = await _storage.read(key: 'access');
      data.headers.remove('requireAuth');
      data.headers['Authorization'] = 'Bearer $accessToken';
      data.headers["Content-Type"] = "application/json";
    }
    print(data.body);
    return data;
  }

  @override
  Future<ResponseData> interceptResponse({ResponseData data}) async {
    print(data.body);
    return data;
  }
}

Client _client = HttpClientWithInterceptor.build(
  interceptors: [
    HttpInterceptor(),
  ],
  retryPolicy: ExpiredTokenRetryPolicy(),
);

Client get client => _client;
```

Also how do I set this up for the exceptions to be handled globally

**Comment by CodingAleCR** [comment]
Hi @patrickoramah The way you are attempting to catch the SocketException should be just fine, in this case the problem is related to the library; it seems like there's a bug with the current version, I am currently checking it out and fixing it soon.

**Comment by CodingAleCR** [comment]
Hi, me again, fixed in #35 with the new release 0.3.2 🚀
